File: go/easy/rna-transcription.go
// https://exercism.org/tracks/go/exercises/rna-transcription
package easy

func ToRNA(dna string) string {
	rna := ""
	for _, nucleotide := range dna {
		switch nucleotide {
		case 'G':
			rna += "C"
		case 'C':
			rna += "G"
		case 'T':
			rna += "A"
		case 'A':
			rna += "U"
		}
	}

	return rna
}

File: ts/easy/rna-transcription.ts
// https://exercism.org/tracks/typescript/exercises/rna-transcription

export function toRna(dna: string): string {
  let rna = "";

  for (let i = 0; i < dna.length; i++) {
    switch (dna[i]) {
      case "G":
        rna += "C";
        break;
      case "C":
        rna += "G";
        break;
      case "T":
        rna += "A";
        break;
      case "A":
        rna += "U";
        break;
      default:
        throw new Error("Invalid input DNA.");
    }
  }

  return rna;
}
